### PR TITLE
Rg/16 create mobile bottom navigation

### PR DESCRIPTION
**Pull request by ranegray**
I created a BottomNav component, added Hero Icons and Tailwind to style it.

### CODE REVIEW

File: lib/Components/BottomNav/BottomNavItem.jsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const BottomNavItem = ({ icon, name, setActive, selected }) => {
+  return (
+    <div
+    onClick={setActive}
+      className={`flex items-center justify-center flex-col ${
+        selected ? "text-selected underline" : ""
+      }`}
+    >
+      {icon}
+      <div className="pointer-events-none">{name}</div>
```
Comment (toreylittlefield) [inline lib/Components/BottomNav/BottomNavItem.jsx:10]: you don't want to show point events?

File: lib/Components/BottomNav/BottomNavItem.jsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const BottomNavItem = ({ icon, name, setActive, selected }) => {
+  return (
+    <div
```
Comment (toreylittlefield) [inline lib/Components/BottomNav/BottomNavItem.jsx:3]: use semantic HTML and get browser & other accessibility benefits by going with a `<button />`

File: lib/Components/BottomNav/BottomNavItem.jsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    onClick={setActive}
+      className={`flex items-center justify-center flex-col ${
+        selected ? "text-selected underline" : ""
```
Comment (toreylittlefield) [inline lib/Components/BottomNav/BottomNavItem.jsx:6]: nice you got this selected state!

File: lib/Components/BottomNav/BottomNavItem.jsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+      }`}
+    >
+      {icon}
```
Comment (toreylittlefield) [inline lib/Components/BottomNav/BottomNavItem.jsx:9]: Can we use React.Children instead or maybe a prop to select the correct icon? That way we don't have to pass in a component as a prop.

File: lib/Components/BottomNav/BottomNav.jsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        icon={<HomeIcon className="h-6 w-6 pointer-events-none" />}
+        name={"Home"}
+      />
+      <BottomNavItem
+        setActive={() => setActive(1)}
+        selected={active === 1 ? true : false}
+        icon={<MagnifyingGlassIcon className="h-6 w-6 pointer-events-none" />}
+        name={"Search"}
+      />
+      <div></div>
```
Comment (toreylittlefield) [inline lib/Components/BottomNav/BottomNav.jsx:28]: empty div

File: lib/Components/BottomNav/BottomNav.jsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      <div className="rounded-full fixed bottom-0 h-24 w-24 flex items-center justify-center bg-white">
```
Comment (toreylittlefield) [inline lib/Components/BottomNav/BottomNav.jsx:29]: again great to use a `<button/>` here

File: pages/index.tsx [hunk 7]
```diff
@@ -18,8 +18,9 @@ export default function Home() {
         <meta name="viewport" content="width=device-width, initial-scale=1" />
         <link rel="icon" href="/favicon.ico" />
       </Head>
-      <main className="bg-background text-foreground">
+      <main className="relative bg-background text-foreground">
```
Comment (toreylittlefield) [inline pages/index.tsx:21]: was the `relative` needed in the end?

File: lib/Components/BottomNav/BottomNav.jsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+const BottomNav = () => {
+  const [active, setActive] = useState(0);
+
+  return (
+    <div className="absolute bottom-0 z-[9999] w-full h-16 text-slate-400 rounded-t-2xl p-1 grid grid-cols-5 justify-items-center items-center bg-white">
+      <BottomNavItem
+        setActive={() => setActive(0)}
+        selected={active === 0 ? true : false}
+        icon={<HomeIcon className="h-6 w-6 pointer-events-none" />}
+        name={"Home"}
```
Comment (toreylittlefield) [inline lib/Components/BottomNav/BottomNav.jsx:20]: text size is smaller in the figma spec for the button titles

File: lib/Components/BottomNav/BottomNav.jsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+        setActive={() => setActive(1)}
+        selected={active === 1 ? true : false}
+        icon={<MagnifyingGlassIcon className="h-6 w-6 pointer-events-none" />}
+        name={"Search"}
+      />
+      <div></div>
+      <div className="rounded-full fixed bottom-0 h-24 w-24 flex items-center justify-center bg-white">
+        <div className="rounded-full h-20 w-20 flex items-center justify-center bg-purple-500 ">
+          <div className="rounded-full h-16 w-16 flex items-center justify-center bg-slate-300">
+            <PlusCircleIcon className="h-6 w-6 text-purple-500" />
```
Comment (toreylittlefield) [inline lib/Components/BottomNav/BottomNav.jsx:32]: text-primary for the text-color

File: lib/Components/BottomNav/BottomNav.jsx [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  UserIcon,
+  PlusCircleIcon,
+  MapPinIcon,
+} from "@heroicons/react/24/outline";
+
+const BottomNav = () => {
+  const [active, setActive] = useState(0);
+
+  return (
+    <div className="absolute bottom-0 z-[9999] w-full h-16 text-slate-400 rounded-t-2xl p-1 grid grid-cols-5 justify-items-center items-center bg-white">
```
Comment (toreylittlefield) [inline lib/Components/BottomNav/BottomNav.jsx:15]: in the figma spec I used `text-foreground/30` for the "faded" unselected buttont ext

File: lib/Components/BottomNav/BottomNav.jsx [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+      <BottomNavItem
+        setActive={() => setActive(1)}
+        selected={active === 1 ? true : false}
+        icon={<MagnifyingGlassIcon className="h-6 w-6 pointer-events-none" />}
+        name={"Search"}
+      />
+      <div></div>
+      <div className="rounded-full fixed bottom-0 h-24 w-24 flex items-center justify-center bg-white">
+        <div className="rounded-full h-20 w-20 flex items-center justify-center bg-purple-500 ">
+          <div className="rounded-full h-16 w-16 flex items-center justify-center bg-slate-300">
```
Comment (toreylittlefield) [inline lib/Components/BottomNav/BottomNav.jsx:31]: again look at the figma spec for the color

File: lib/Components/BottomNav/BottomNavItem.jsx (hunk 1 above)
Comment (ranegray) [reply lib/Components/BottomNav/BottomNavItem.jsx:10]: I suppose I can get get rid of those now. They were a step in my problem solving process lol

File: lib/Components/BottomNav/BottomNavItem.jsx (hunk 3 above)
Comment (ranegray) [reply lib/Components/BottomNav/BottomNavItem.jsx:6]: i wish there was a crying emoji

File: lib/Components/BottomNav/BottomNav.jsx (hunk 5 above)
Comment (ranegray) [reply lib/Components/BottomNav/BottomNav.jsx:28]: That empty div is how I positioned the icons. I took the center div out of the normal flow with a fixed position and I put an empty div to ensure proper spacing.

File: pages/index.tsx (hunk 7 above)
Comment (ranegray) [reply pages/index.tsx:21]: mmmm i'll check it out see if i can't do without it.

**Comment by Netdreion** [comment]
looks good to me

File: lib/Components/BottomNav/BottomNav.jsx (hunk 6 above)
Comment (ranegray) [reply lib/Components/BottomNav/BottomNav.jsx:29]: @toreylittlefield should i use a button one each nested div for center button or just the top level div?

**Comment by ranegray** [comment]
Okay, I made the changes and merged main into this branch. All should be right in the world. I fixed styling, semantics and some convention things. Let me know if there are any other improvements we want to make to this branch before merge.

**Comment by toreylittlefield** [comment]
@ranegray just resolve the conflicts and I'll review after 💯  nice work on this

**Comment by ranegray** [comment]
Resolved conflicts, ready for review.

**Review by toreylittlefield — APPROVED** [review]
@ranegray well done 🥇 

I have a few improves that I think can be cleaned up later but this is great. Excellent work matching the code to the design spec!

**Pull request merged**